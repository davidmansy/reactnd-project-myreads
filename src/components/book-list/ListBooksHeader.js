import React from 'react';

function ListBooksHeader() {
  return (
    <div className="list-books-title">
      <h1>MyReads</h1>
    </div>
  );
}

export default ListBooksHeader;
