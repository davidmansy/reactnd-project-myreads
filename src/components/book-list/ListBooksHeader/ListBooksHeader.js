import React from 'react';
import PropTypes from 'prop-types';

function ListBooksHeader(props) {
  return (
    <div className="list-books-title">
      <h1>{props.title}</h1>
    </div>
  );
}

ListBooksHeader.propTypes = {
  title: PropTypes.string.isRequired
};

export default ListBooksHeader;
