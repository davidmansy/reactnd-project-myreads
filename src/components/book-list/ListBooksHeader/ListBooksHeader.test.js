import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import ListBooksHeader from './ListBooksHeader';
import { MemoryRouter } from 'react-router-dom';
import { cleanup } from 'react-testing-library';

afterEach(cleanup);

describe('ListBooksHeader', () => {
  const props = {
    title: 'title'
  };

  it('renders', () => {
    const container = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ListBooksHeader {...props} />
      </MemoryRouter>,
      container
    );
  });

  it('snapshots', () => {
    const testRenderer = TestRenderer.create(
      <MemoryRouter>
        <ListBooksHeader {...props} />
      </MemoryRouter>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
