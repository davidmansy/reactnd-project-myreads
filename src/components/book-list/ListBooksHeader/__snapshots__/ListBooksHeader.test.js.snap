// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBooksHeader snapshots 1`] = `
<div
  className="list-books-title"
>
  <h1>
    title
  </h1>
</div>
`;
